Ticket #36026 WMS Integration Discussion

mithunporandlajkt opened the pull request:
api\SendCustomer.p
api\SendCustomerRequestHandler0001.p
api\SendRelease.p
api\SendReleaseRequestHandler0001.p
api\SendFinishedGood.p
api\SendFinishedGoodRequestHandler0001.p
api\SendPurchaseOrder.p
api\SendPurchaseOrderRequestHandler0001.p
api\SendVendor.p
api\SendVendorRequestHandler0001.p
api\APIOutboundEventsViewer.w
api\APIOutboundTest.w
api\CallOutBoundAPI.p
api\PrepareOutboundRequest.p
api\ResponseDataViewer.w
api\ResponseHandler0001.p
api\JSONProcs.p
api\PrepareAndCallOutboundRequest.p
api\CreateAPIOutboundEvent.p
api\ttArgs.i
testers\OutboundAPITester.p
viewers\cust.w
viewers\itemfg.w
viewers\vend.w
est\custfly.w
oerep\r-relprt.w
po\d-poordl.w
system\openlookup.p
windows\l-lookup.w

mithunporandlajkt committed:
Files:
api/AddProduct.p
api/AddPurchaseOrder.p
api/AddVendor.p
api/CallOutBoundAPI.p
api/OutboundEvent.w
api/ResponseDataViewer.w
testers/OutboundAPITester.p

mithunporandlajkt committed:
Files:
api/AddCustomer.p
api/AddPicklist.p
api/AddProduct.p
api/AddPurchaseOrder.p
api/AddOrder.p
api/CallOutboundAPI.p
api/ResponseDataViewer.w
testers/OutboundAPITester.p
api/APIOutboundEventsViewer.w
api/PrepareOutboundRequest.p
api/premier/AddCustomerRequestHandler.p
api/premier/AddProductRequestHandler.p
api/premier/AddVendorRequestHandler.p
api/premier/AddPurchaseOrderRequestHandler.p
api/premier/AddPicklistRequestHandler.p
api/premier/ResponseHandler

mithunporandlajkt committed:
Deleting File:
api/SingleJSONResponse.p

mithunporandlajkt committed:
Ticket #36026 WMS Integration Discussion

Files:
api/ResponseDataViewer.w
api/APIOutboundEventsViewer.w
api/PrepareOutboundRequest.p
api/AddCustomerRequestHandler0001.p
api/AddProductRequestHandler0001.p
api/AddVendorRequestHandler0001.p
api/AddPurchaseOrderRequestHandler0001.p
api/AddPicklistRequestHandler0001.p
api/ResponseHandler0001.p
api/APIOutboundTest.w

mithunporandlajkt committed:
Ticket #36026 WMS Integration Discussion

Files:
api\AddCustomer.p
api\AddCustomerRequestHandler0001.p
api\AddPicklist.p
api\AddPicklistRequestHandler0001.p
api\AddProduct.p
api\AddProductRequestHandler0001.p
api\AddPurchaseOrder.p
api\AddPurchaseOrderRequestHandler0001.p
api\AddVendor.p
api\AddVendorRequestHandler0001.p
api\APIOutboundEventsViewer.w
api\APIOutboundTest.w
api\CallOutBoundAPI.p
api\PrepareOutboundRequest.p
api\ResponseDataViewer.w
api\ResponseHandler0001.p
api\JSONProcs.p
api\PrepareAndCallOutboundRequest.p
api\ttArgs.i
testers\OutboundAPITester.p
viewers\cust.w
viewers\itemfg.w
viewers\vend.w
est\custfly.w
oerep\r-relprt.w
po\d-poordl.w

mithunporandlajkt committed:
Ticket #36026 WMS Integration Discussion

Files:
api\AddPicklist.p
api\AddPicklistRequestHandler0001.p
api\APIOutboundEventsViewer.w
api\APIOutboundTest.w
po\d-poordl.w
system\openlookup.p
windows\l-lookup.w

mithunporandlajkt committed:
Merge branch 'develop' into feature/36026_eWMS_Integration_Discussion

mithunporandlajkt committed:
Ticket #36026 WMS Integration Discussion

Files:
api\SendCustomer.p
api\SendCustomerRequestHandler0001.p
api\SendRelease.p
api\SendReleaseRequestHandler0001.p
api\SendFinishedGood.p
api\SendFinishedGoodRequestHandler0001.p
api\SendPurchaseOrder.p
api\SendPurchaseOrderRequestHandler0001.p
api\SendVendor.p
api\SendVendorRequestHandler0001.p
api\APIOutboundEventsViewer.w
api\APIOutboundTest.w
api\CallOutBoundAPI.p
api\PrepareOutboundRequest.p
api\ResponseDataViewer.w
api\ResponseHandler0001.p
api\JSONProcs.p
api\PrepareAndCallOutboundRequest.p
api\ttArgs.i
testers\OutboundAPITester.p
viewers\cust.w
viewers\itemfg.w
viewers\vend.w
est\custfly.w
oerep\r-relprt.w
po\d-poordl.w
system\openlookup.p
windows\l-lookup.w

mithunporandlajkt committed:
Files:

api/CallOutBoundAPI.p

brad-advantzware reviewed (changes requested):
Looks generally good.
Changes:  
Remove the calls to outbound within the existing application programs (cust.w, d-poordl.w, etc.)
Convert the internal Procedures in those programs to a singular external program to encapsulate the ttArgs definition and for us to add our license and security controls in one place.
REmove any UI elements inside a .p (MESSAGE) - parameterize these to return to caller for the caller to determine whether to present the error or not.

mithunporandlajkt committed:
Files:
api\SendFinishedGood.p
api\SendFinishedGoodRequestHandler0001.p
api\SendPurchaseOrder.p
api\SendPurchaseOrderRequestHandler0001.p
api\APIOutboundTest.w
api\PrepareAndCallOutboundRequest.p
viewers\cust.w
viewers\itemfg.w
viewers\vend.w
est\custfly.w
oerep\r-relprt.w
po\d-poordl.w
system\openlookup.p
